### PR TITLE
[2201.5.x] Fix closure variables related compiler crash inside do clause in query expression

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fix closure variables related compiler crash inside do clause in query expression

master PR - https://github.com/ballerina-platform/ballerina-lang/pull/40014/files

### CODE REVIEW

**Comment by pcnfernando** [comment]
Seems the query closure-related tests are failing in 5.x as some specific changes in query implementations are not upto date

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -5951,6 +5951,9 @@ public void visit(BLangSimpleVarRef varRefExpr) {
             BVarSymbol varSymbol = (BVarSymbol) varRefExpr.symbol;
             if (varSymbol.originalSymbol != null) {
                 varRefExpr.symbol = varSymbol.originalSymbol;
+                if (varSymbol.closure) {
```
Comment (SasinduDilshara) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:5954]:
```suggestion
                if varSymbol.closure {
```

**Comment by SasinduDilshara** [comment]
> Seems the query closure-related tests are failing in 5.x as some specific changes in query implementations are not upto date

Shall we close this PR for now and create a new issue for this?

**Comment by pcnfernando** [comment]
> 

I think since this is fixed on later versions, no need to create a new issue. Mention the scenario and close the PR

**Comment by SasinduDilshara** [comment]
Closing this Pr since the query related changes are not up-to-date in here.

The fix was added to the 6.x and 7.x

**Pull request closed without merging**